src/js/utils/ServiceUtil.js

```diff
@@ -451,55 +451,23 @@ const ServiceUtil = {
     return getFindPropertiesRecursive(service, schema.properties);
   },
 
-  getAppDefinitionFromService(service) {
-
-    let appDefinition = JSON.parse(JSON.stringify(service));
-
-    // General
-    appDefinition.id = service.getId();
-    appDefinition.cpus = service.getCpus();
-    appDefinition.mem = service.getMem();
-    appDefinition.disk = service.getDisk();
-    appDefinition.instances = service.getInstancesCount();
-    appDefinition.cmd = service.getCommand();
-
-    // Optional
-    appDefinition.executor = service.getExecutor();
-    appDefinition.fetch = service.getFetch();
-    appDefinition.constraints = service.getConstraints();
-    appDefinition.acceptedResourceRoles = service.getAcceptedResourceRoles();
-    appDefinition.user = service.getUser();
-    appDefinition.labels = service.getLabels();
-    appDefinition.healthChecks = service.getHealthChecks();
-
-    let containerSettings = service.getContainerSettings();
-    if (containerSettings &&
-      ((containerSettings.docker && containerSettings.docker.image) ||
-      containerSettings.type === VolumeConstants.type.MESOS)
-    ) {
-      appDefinition.container = containerSettings;
-
-      if (appDefinition.container.type === VolumeConstants.type.MESOS) {
-        delete(appDefinition.container.docker);
-      }
-    }
+  getDefinitionFromSpec(spec) {
+    let definition = spec.toJSON();
 
-    appDefinition.updateStrategy = service.getUpdateStrategy();
-    appDefinition.residency = service.getResidency();
-    appDefinition.ipAddress = service.getIpAddress();
+    if (spec instanceof ApplicationSpec) {
 
-    // Environment Variables
-    appDefinition.env = service.getEnvironmentVariables();
+      // TODO: It looks that this hook operates over Application (Service)
```

Comment:
ok, I will pass down the `spec`. Where can I find the plug-ins that might be affected? -- If any